pkg/reconciler/addressableservice/addressableservice.go

```diff
@@ -18,115 +18,49 @@ package addressableservice
 
 import (
 	"context"
-	"reflect"
 
-	"go.uber.org/zap"
-	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/equality"
 	apierrs "k8s.io/apimachinery/pkg/api/errors"
 	corev1listers "k8s.io/client-go/listers/core/v1"
-	"k8s.io/client-go/tools/cache"
-	"k8s.io/client-go/tools/record"
 	"knative.dev/pkg/apis"
 	duckv1beta1 "knative.dev/pkg/apis/duck/v1beta1"
-	"knative.dev/pkg/controller"
 	"knative.dev/pkg/logging"
 	"knative.dev/pkg/network"
 	"knative.dev/pkg/tracker"
-	"knative.dev/sample-controller/pkg/apis/samples/v1alpha1"
-	clientset "knative.dev/sample-controller/pkg/client/clientset/versioned"
-	listers "knative.dev/sample-controller/pkg/client/listers/samples/v1alpha1"
+	v1alpha1 "knative.dev/sample-controller/pkg/apis/samples/v1alpha1"
 )
 
 // Reconciler implements controller.Reconciler for AddressableService resources.
 type Reconciler struct {
-	// Client is used to write back status updates.
-	Client clientset.Interface
+	Core
 
 	// Listers index properties about resources
-	Lister        listers.AddressableServiceLister
 	ServiceLister corev1listers.ServiceLister
```

Comment:
Is this used for anything? In general, is the idea that you list all the things you rely on here? It's probably answered in the design doc.

Reply:
This is everything that is custom to your reconciler that is not already included in `Core`